src/ast-utils.ts

```diff
@@ -0,0 +1,78 @@
+import { AST, SourceCode } from 'eslint';
+import { Node } from 'estree';
+
+export const isSemicolonToken = (token: AST.Token): boolean =>
+  token.value === ';' && token.type === 'Punctuator';
+
+export const isTokenOnSameLine = (
```

Comment:
Any interest in writing some unit tests for these utils. It isn't necessary as we have integration tests for all the rules - just a nice to have.